enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,130 @@
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC resources. This enhancement proposes adding support for Prometheus `ServiceMonitors` and `PrometheusRules` resources:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+OLM would then need to be updated to identify the `ServiceMonitors` and `PrometheusRules` resources and deployt hem as part of an `InstallPlan`. There is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the appropriate RBAC privileges to view the `ServiceMonitor` and `PrometheusRule` resource events in the namespace that the operator is being deployed to. As such, the cluster admin will need to configure the Prometheus Operator to watch for events in the correct namespace.
+
+### OpenShift Monitoring Prometheus Operator Support
+
+#### Requirements
+
+Once OLM offers bundle support for the `ServiceMonitor` and `PrometheusRule` resources, a sanctioned set of OLM managed operators should be able to report metrics to the OpenShift Monitoring Prometheus Operator.
+
+A number of requirements must be met for an applications metrics to be scraped by the OpenShift Monitoring Prometheus Operator.
+
+##### Namespace Requirements
+
+The application must be deployed in a namespace that:
+
+- Is prefixed with `openshift-`
+- Is labeled with `openshift.io/cluster-monitoring=true`
+
+##### ServiceMonitor Requirements
+
+A `ServiceMonitor` object must be created in the namespace mentioned above and point to the application's metrics endpoint.
+
+>Note: This requirement is already fulfilled with OLM bundle support for Prometheus resources.
+
+##### RBAC Requirements
+
+A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitor` and `PrometheusRule` resources in the newly created namespace.
+
+#### Fulfilling Namespace and RBAC Requirements
+
+OLM itself will not be responsible for fulfilling the Namespace and RBAC requirements. Instead, OLM will define how operator authors can provide OpenShift Console with the information required to fulfill these requirements.
+
+OLM [building-your-csv](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md) documentation will be updated to describe two new supported annotations:
+
+- The `suggested-namespace` annotation. When the `suggested-namespace` annotation is present the UI will highlight the suggested namespace when installing the operator. OLM itself will not require that the operator is deployed in the namespace defined by the `suggested-namespace` annotation.
```

Comment:
these annotations should be namespaced as well - probably operatorframwork.io or something